src/main.rs

```diff
@@ -27,41 +27,53 @@ mod util;
 #[subxt(runtime_metadata_path = "invarch_metadata.scale")]
 pub mod invarch {}
 
-pub async fn set_repo(
+/// Gets Repo Data file from IP Set and returns a `RepoData` struct created from the file.
+/// If no file, return `RepoData` struct with defaults
+pub async fn get_repo(
     ips_id: u32,
     api: invarch::RuntimeApi<DefaultConfig, PolkadotExtrinsicParams<DefaultConfig>>,
 ) -> BoxResult<RepoData> {
+    // Setup IPFS client
     let mut ipfs_client = IpfsClient::default();
+
+    // Get assets (IPFs, NFTs, nested IPS, NFT collections) under `ips_id` IP Set
     let data = api
         .storage()
-        .inv4()
-        .ip_storage(&ips_id, None)
+        .inv4() // pallet
+        .ip_storage(&ips_id, None) // storage entity
         .await?
         .ok_or(format!("Ips {ips_id} does not exist"))?
         .data
         .0;
 
+    // Find "RepoData" IPF and return `RepoData` struct if it exists
     for file in data {
         if let AnyId::IpfId(id) = file {
             let ipf_info = api
                 .storage()
-                .ipf()
-                .ipf_storage(&id, None)
+                .ipf() // pallet
+                .ipf_storage(&id, None) // storage entity
                 .await?
                 .ok_or("Internal error: IPF listed from IPS does not exist")?;
             if String::from_utf8(ipf_info.metadata.0.clone())? == *"RepoData" {
                 return RepoData::from_ipfs(ipf_info.data, &mut ipfs_client).await;
             }
         }
     }
+
+    // Return default `RepoData` if file doesn't exist
     Ok(RepoData {
         refs: Default::default(),
         objects: Default::default(),
     })
 }
 
+/// Git will call this helper program because it does not natively support git-remote-inv4
 #[tokio::main]
 async fn main() -> BoxResult<()> {
+    // Setup config file
+
```

Comment:
```suggestion
```

Reply:
I guess you said this is going to go aways anyway